Attribute Analysis.ipynb

```diff
@@ -602,6 +606,336 @@
     "display(ui, out)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# (M2) Metric: Object occurrences and cooccurrences\n",
+    "<a id=\"metric2\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "<a id=\"metric2_setup\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "categories = dataset.categories\n",
+    "names = dataset.labels_to_names\n",
+    "counts = pickle.load(open(\"results/{}/2.pkl\".format(folder_name), \"rb\"))\n",
+    "topn = 10\n",
+    "counts_mat = [np.zeros((len(categories), len(categories))) for i in range(num_attrs)]\n",
+    "for key in counts[0].keys():\n",
+    "    a, b = key.split('-')\n",
+    "    a, b = int(a), int(b)\n",
+    "    for count in range(num_attrs):\n",
+    "        counts_mat[count][b][a] = counts[count][key]\n",
+    "        counts_mat[count][a][b] = counts[count][key]\n",
+    "instance_counts = [np.diagonal(counts_mat[i]) for i in range(num_attrs)]\n",
+    "normalized = [np.divide(counts_mat[i], instance_counts[i]) for i in range(num_attrs)]\n",
+    "            \n",
+    "def make_dist(counts, total):\n",
+    "    samples = np.zeros(total)\n",
+    "    samples[:int(counts)] = 1\n",
+    "    return samples\n",
+    "# Looking at if the number of times an object appears with different attributes is statistically significant for some specific attr (analyze_attr)\n",
```

Comment:
I don't understand the end of this print statement (what does analyze_attr refer to?)